src/distilabel/steps/task/evol_quality/base.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+if sys.version_info < (3, 11):
+    from enum import EnumMeta as EnumType
+else:
+    from enum import EnumType
+
+from typing import TYPE_CHECKING, Any, Dict, List, Union
+
+import numpy as np
+from pydantic import Field
+from typing_extensions import override
+
+from distilabel.steps.base import RuntimeParameter
+from distilabel.steps.task.base import Task
+from distilabel.steps.task.evol_quality.utils import MutationTemplates
+from distilabel.steps.task.typing import ChatType
+
+if TYPE_CHECKING:
+    from distilabel.steps.base import StepInput
+    from distilabel.steps.typing import StepOutput
+
+
+class EvolQuality(Task):
+    """
+    What Makes Good Data for Alignment? A Comprehensive Study of Automatic Data Selection in Instruction Tuning
+    and
+    WizardLM: Empowering Large Language Models to Follow Complex Instructions
+
+    Reference:
+        - https://arxiv.org/abs/2312.15685
+        - https://arxiv.org/abs/2304.12244
+        - https://github.com/h2oai/h2o-wizardlm
+        - https://github.com/nlpxucan/WizardLM/Evol_Instruct
+
+    Runtime parameters:
+
+    - `seed`: The number of evolutions to be run.
+
+    Columns:
+
+    - `input`: instruction
+    - `output`: there's multiple scenarios:
+        - `store_evolutions=False`, `generate_answers=False` -> (evolved_response, model_name)
+        - `store_evolutions=True`, `generate_answers=False` -> (evolved_responses, model_name)
```

Comment:
We agreed to the following format for documenting the columns: https://github.com/argilla-io/distilabel/blob/66063e2a6f51ae7354d4bf4ac2f4aca4131abe38/src/distilabel/steps/task/quality_scorer.py#L49-L54